### PR TITLE
Support PG types in schemeshard

### PR DESCRIPTION
**Pull request by lex007in**
### Changelog entry 

* Add test for shard splitting by PG key
* Support PG types for PartitionKeySchema in PQ

### Changelog category 

* Improvement

### Additional information

...

### CODE REVIEW

File: ydb/core/tx/schemeshard/schemeshard__table_stats_histogram.cpp [hunk 1]
```diff
@@ -6,7 +6,6 @@ namespace NKikimr {
 namespace NSchemeShard {
 
 static bool IsIntegerType(NScheme::TTypeInfo type) {
-    // TODO: support pg types
```
Comment (lex007in) [inline ydb/core/tx/schemeshard/schemeshard__table_stats_histogram.cpp:9]: Generic not-only-integer realization is enough. Tested in SplitShardsWhithPgKey test.

**Review by azevaykin — DISMISSED** [review]

File: ydb/core/tx/schemeshard/ut_split_merge/ut_split_merge.cpp [hunk 2]
```diff
@@ -158,6 +158,84 @@ Y_UNIT_TEST_SUITE(TSchemeShardSplitBySizeTest) {
...
+                           {NLs::PartitionCount(1)});
+
+        TestAlterTable(runtime, ++txId, "/MyRoot", R"(
+                        Name: "Table"
+                        PartitionConfig {
+                            PartitioningPolicy {
+                                MinPartitionsCount: 100
+                                MaxPartitionsCount: 100
+                                SizeToSplit: 1
+                                FastSplitSettings {
```
Comment (ijon) [inline ydb/core/tx/schemeshard/ut_split_merge/ut_split_merge.cpp:203]: Use of `FastSplitSettings` was removed from split-merge code quite some time ago. So it is meaningless to specify them in tests.

There are still some not updated tests (and they surprisingly continue to work ok), but new tests should not employ `FastSplitSettings`.

File: ydb/core/tx/schemeshard/schemeshard__table_stats_histogram.cpp (hunk 1 above)
Comment (ijon) [reply ydb/core/tx/schemeshard/schemeshard__table_stats_histogram.cpp:9]: Its enough to work but not enough to work well.
Below in the same file `IsIntegerType()` is used to choose between fast and slow branches for split prefix search. And failing to support integer pg types as integer forces `ChooseSplitKeyByHistogram()` to choose the slow pass.

**Review by ijon — CHANGES_REQUESTED** [review]

File: ydb/core/tx/schemeshard/schemeshard__table_stats_histogram.cpp (hunk 1 above)
Comment (lex007in) [reply ydb/core/tx/schemeshard/schemeshard__table_stats_histogram.cpp:9]: Yes, support for this optimization requires the ability to increment a value of integer pg type which is not  currently supported. I'll do it in separate issue: https://github.com/ydb-platform/ydb/issues/11168

**Review by ijon — APPROVED** [review]

**Pull request merged**